vizro-core/tests/tests_utils/asserts.py

```diff
@@ -4,27 +4,31 @@
 import plotly
 
 
-def strip_keys(object, keys):
-    """Strips all entries with key "id" from a dictionary, regardless of how deeply it's nested.
-
-    This makes it easy to compare dictionaries generated from Dash components we've created that contain random IDs.
-    """
+def _strip_keys(object, keys):
+    """Strips from a JSON object all entries where the key is in keys, regardless of how deeply it's nested."""
     if isinstance(object, dict):
-        object = {key: strip_keys(value, keys) for key, value in object.items() if key not in keys}
+        object = {key: _strip_keys(value, keys) for key, value in object.items() if key not in keys}
     elif isinstance(object, list):
-        object = [strip_keys(item, keys) for item in object]
+        object = [_strip_keys(item, keys) for item in object]
     return object
 
 
-def component_to_dict(component: dash.development.base_component.Component) -> dict:
+def _component_to_dict(component: dash.development.base_component.Component) -> dict:
+    """Prepares a Dash component for comparison by conversion to JSON object."""
     return json.loads(json.dumps(component, cls=plotly.utils.PlotlyJSONEncoder))
 
 
-# TODO: implement some sort of depth limit to comparison so can use in high level tests, roll out more widely across
-# tests
 def assert_component_equal(left, right, keys_to_strip=None):
```

Comment:
Given that this labelled as a public function, shall we also add it to the docs?

Reply:
This is not a public function because it's not part of the public `vizro` package. It's "public" only to developers who write tests for vizro.

Possibly in future we would have some public API `from vizro.testing import assert_component_equal` (like Dash and pandas) if vizro _users_ would find it useful, e.g. to write tests for their custom components. This PR doesn't do anything like that though - all the code here lives in `tests`, not in `src`.

Reply:
Ooops, I overlooked that its in the `tests` folder. Nevermind then 👍